src/main/java/com/synopsys/integration/detect/ApplicationUpdater.java

```diff
@@ -589,16 +573,38 @@ protected ProxyInfo getProxyInfo() {
         return proxyInfo;
     }
 
-    private Response download(File installDirectory, HttpUrl downloadUrl, String currentVersion) throws IntegrationException, IOException {
+    private File download(File installDirectory, HttpUrl downloadUrl, String currentVersion) throws IntegrationException, IOException {
         logger.info("{} Checking {} API for centrally managed Detect version to download to {}.", LOG_PREFIX, downloadUrl, installDirectory.getAbsolutePath());
         final Map <String, String> headers = new HashMap<>();
         headers.put(DOWNLOAD_VERSION_HEADER, currentVersion);
         final Request request = new Request(downloadUrl, HttpMethod.GET, null, new HashMap<>(), headers, null);
         ProxyInfo proxyInfo = getProxyInfo();
         final IntHttpClient intHttpClient = getIntHttpClient(proxyInfo);
         try (final Response response = intHttpClient.execute(request)) {
-            return response;
+            return handleResponse(response, currentVersion, installDirectory);
+        }
+    }
+    
+    private File handleResponse(Response response, String currentInstalledVersion, File installDirectory) throws IOException, IntegrationException {
```

Comment:
Can we please add a method description with what return values (including null) indicates either here or at installOrUpdateScanner()/handleSuccessResponse()/wherever you feel it would serve the most value? That way the next time we look at this code it will be immediately clear rather than having to walk through installOrUpdateScanner()>download()>handleResponse()>validateDownloadedJar()  to understand all the possible scenarios/exceptions/etc.

Reply:
Added a note that there are only two possible return values - an instance or a null. Thanks.